### DISCUSSION TITLE
Best way to add media for 15th April | Memorial?

### DISCUSSION THREAD

**Discussion by Chipperchap** [sircharlo/meeting-media-manager#242] [Q&A] [answered]
As there is no midweek meeting material next week an error shows in JWMMF which makes sense, also the folder does not appear for the midweek meeting - again that is expected behaviour. Please, however, what would you recommend is the best way to add media such as the songs and any images the speaker wishes to use for the evening of the 15th April?  Would this need to be done manually?

UPDATE # 1:  ...so I created a folder with the date in my JWMMF folder and it now shows in the media player with the media I have placed in there - but the date does not appear in cloud folder nor sync with other users in the congregation. (I am VO).

UPDATE # 2: I expect that the week of the C.O will be similar in that our regular midweek meeting is on Thursdays, but the C.O meeting is on the Tuesday - please what is the suggestion of how this is best handled by JWMMF?

UPDATE # 3: I suppose I could change the meeting time in the settings to the Friday temporarily and use that to sync.

**Comment by sircharlo** [comment, marked as answer]
Good question!
With today's release, if a folder already exists, is a valid date, and falls between the monday-sunday period selected, then you'll be able to add custom media to it, even if the specific folder's date doesn't fall on a typical meeting date.
Does that make sense?

**Reply by sircharlo** [reply]
Also, what you said in your update number 3 is what I had been doing for our congregation locally... But now there will be more flexibility, depending on what's easier for the end-user.

**Reply by Chipperchap** [reply]
Thank you for the replies and hard work.
I regret to inform you however that I cannot see any changes nor make anything effectively work for me.
Process:

- Update JWMMF to latest version on both computers - home and K-Hall (Via remote desktop)
- Modify midweek meeting to Friday temporarily on home PC.
- Open K-Hall JWMMF on K-Hall PC and it now shows Friday as midweek meeting day - great!

- On home PC, create a folder with proper naming and insert the two memorial songs into there. Put in local JWMMF folder.  Click on "Get Media!" to sync.  Also click on the cloud folder. 
- Cloud folder does not show 'Target meeting' with any 15th folder (2022-04-15). 
- Home PC Media display folder (Alt +D) shows folder for 2022-04-15 with songs - great!
- Check K-Hall PC - no change, no folder, no songs. 
- Check PowerFolder is synchronizing... yup - all good... but no 2022-04-15 folder in JWMMF sync folder.
- Open PowerFolder and manually insert 2022-04-15 folder with songs in it. PowerFolder syncs folder with cloud.  Check K-Hall PC... nothing there. Restart JWMMF... nothing there.  Cache shows (837.6MB) - Home PC shows cache (1190.9MB)
- Allow time for all synchronization to be completed... still nothing... just the error alert saying that there is no material for that week: "An error occurred while fetching media for the midweek meeting. Please... organizer" - so I am thinking that is what is blocking the sync... or am I doing something obviously wrong?

**Reply by sircharlo** [reply]
If using congregation sync, you need to upload the additional media (ie, songs) using the **cloud upload** button, on the main page of JWMMF.
![image](https://user-images.githubusercontent.com/1275666/162550485-9edff254-2695-4b64-a6a5-fefa2d07904e.png)
![image](https://user-images.githubusercontent.com/1275666/162550502-e7fed778-1f17-47ac-b86b-d91dcacb3f95.png)

In my case, I created the "2022-04-15" folder manually in my cloud sync provider's media folder, but you would follow the same procedure if you temporarily change your midweek meeting to Friday. If your files aren't showing up in the cloud sync area, something isn't right.

**Reply by sircharlo** [reply]
After uploading to cloud sync using the cloud upload button, and confirming that you see them in the cloud sync area of your home PC, you should be able to "get media" on the KH computer and sync media from JW.org as well as from the cloud folders. Also, the uploaded files would of course be visible from the cloud upload button of JWMMF on the KH computer...

In any case could you post screenshots?

**Reply by Chipperchap** [reply]
Please check out this video...

https://simonsheena-gmail.tinytake.com/msc/NjcxOTM5NV8xOTMxNDA2NQ

**Reply by sircharlo** [reply]
I think I see what happened. Perhaps I wasn’t clear, I do apologize. This screenshot from your video holds the key, I believe:
![simonsheena-gmail](https://user-images.githubusercontent.com/1275666/162565149-94bea223-37ff-400b-8d1e-8dd9c92d4e96.png)

The "E" folder was created by you, I'm assuming? In any case, JWMMF will *never* see that folder. 
Here's where it looks in congregation sync: 
- [Congregation sync root]/**forcedPrefs.json**: This is the file containing the settings that you chose to sync across computers
- [Congregation sync root]/**Hidden**/[Dated folders]: This is where JWMMF will store information about files that should not be downloaded, or skipped, and therefore not show up in the final media list
- [Congregation sync root]/**Media**/[Dated folders]: This is where JWMMF will store additional and custom media. **This is where your dated 2022-04-15 folder should go.**

Once you create the **2022-04-15** folder under the automatically created [Congregation sync root]/**Media** folder in PowerFolder, then April 15th will show up in the target meeting list.

**Reply by Chipperchap** [reply]
Thank you for your fine assistance.  It is truly appreciated. 

I put the folder in the right place now and it is working. However I don't know where the 'E' comes from.

Latest video results;:

https://simonsheena-gmail.tinytake.com/msc/NjcxOTU1M18xOTMxNDM4OQ

**Reply by sircharlo** [reply]
We'll get there eventually 😂

I can definitely see how that E folder could lead to confusion. 

The "E" folder is created automatically on the local computer. Why? In the case where multiple congregations would use the same Windows account for JWMMF.

Suppose an English congregation and a Spanish congregation use the same Windows account, and the same media output folder (for example, C:\Users\Hall\Videos). Now suppose that they also have overlapping meeting days.

The end result would be that the dated folders created by JWMMF would contain media in two languages, leading to potential confusion during meetings. 
To avoid this, JWMMF always creates a subfolder in the media output folder with the current congregation's language code. Just to keep things tidy.

**Reply by Chipperchap** [reply]
I see - thank you for that - it is starting to get through my slow brain! 😆 According to the official guidelines (S-132) - each language group sharing a K-Hall would have their own login account to the K-Hall PC... but I can see how having a different folder within JWMMF would avoid issues were a congregation not to follow that recommendation.

**Reply by sircharlo** [reply]
As for the music button, this was buried deep in the release notes: 

- Added a confirmation click for background music button, to prevent accidental starting

I'm still struggling with how to visually convey that idea, of "click twice to confirm".

In any case, here's how it works: 
- click the music button once: it becomes orange, meaning "are you sure it's time to play music?". 
- click it again while the button is still orange, and then the music starts. 
- if you don't click it a second time, then the button returns to its inactive state. 

This is done to prevent clicking the music button accidentally during a meeting, and triggering the music. I'm open to suggestions if there's a better or different way.

The same "click twice" logic is also present for congregation deletion (click the delete button twice to confirm), and for stopping a video while in media presentation mode. All in the spirit of avoiding errors from inadvertent clicks.

**Reply by sircharlo** [reply]
> I see - thank you for that - it is starting to get through my slow brain! 😆 According to the official guidelines (S-132) - each language group sharing a K-Hall would have their own login account to the K-Hall PC... but I can see how having a different folder within JWMMF would avoid issues were a congregation not to follow that recommendation.

Agreed, and yet it's happened. Better to have a failsafe in place just in case, I reckon.

**Comment by Chipperchap** [comment]
All questions have been answered:

- Additional meetings such as memorial.  Go into settings and temporarily adjust meeting day to match date of event. Add dated folder to Sync folder. Deploy "Get Media" and "Cloud" buttons. In "Cloud" insert media as needed such as songs and custom images.  On remote PCs open JWMMF and check meeting day has changed and "Get Media" has 'fetched' custom meeting media. 
- Background Music: Click twice. (Now has 'are you sure?' button to prevent accidental use). 

Thank you so much @sircharlo for your help. Outstanding software authorship! 

![Commendation](https://user-images.githubusercontent.com/35364657/162590356-f853b923-af67-4629-a0f6-721982e61d1d.jpg)

**Reply by sircharlo** [reply]
Always a pleasure!
